apps/frontend/src/app/components/auth/Login.tsx

```diff
@@ -25,77 +25,52 @@ import {
   SignInWithGoogleButton,
 } from './styles';
 import { trpc } from '../../../utils/trpc';
-import { useCallback, useContext, useEffect, useState } from 'react';
-import { validateEmail, validatePassword } from '@dnd-assistant/shared-utils';
+import { useContext, useState } from 'react';
 import { getIonInputClassNames } from './input';
 import { SessionContext } from '../../context/session/SessionContext';
 import { Routes } from '../../routes';
+import { handleTRPCErrors } from '../../../utils/handleTRPCErrors';
 
 export const Login: React.FC = () => {
-  const login = trpc.user.login.useMutation();
+  const [presentToast] = useIonToast();
+  const loginMutation = trpc.user.login.useMutation({
+    onError: (error) => {
+      handleTRPCErrors(error.data?.httpStatus, presentToast, {
+        400: 'The email or password you submited is incorrect.',
+      });
+    },
+  });
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
-  const [toastMessage, setToastMessage] = useState('');
   const [emailIsTouched, setEmailIsTouched] = useState(false);
   const [passwordIsTouched, setPasswordIsTouched] = useState(false);
-  const [emailIsValid, setEmailIsValid] = useState(true);
-  const [passwordIsValid, setPasswordIsValid] = useState(true);
-  const [loginIsDisabled, setLoginIsDisabled] = useState(false);
-  const [showToast, setShowToast] = useState(false);
 
   const { setSession } = useContext(SessionContext);
   const router = useIonRouter();
 
-  const loginHook = useCallback(() => {
-    setLoginIsDisabled(true);
-    login.mutate({
+  const loginHook = () => {
+    loginMutation.mutate({
       email,
       password,
     });
-  }, [email, password, login]);
-
-  useEffect(() => {
-    if (login.isSuccess) {
-      setSession(login.data);
-      router.push(Routes.Dashboard);
-    } else if (login.isError) {
-      setShowToast(true);
-    }
-    setLoginIsDisabled(false);
-  }, [login.data, login.isSuccess, login.isError, setSession, router]);
+  };
 
-  useEffect(() => {
-    if (login.error) {
-      const statusCode = login.error?.data?.httpStatus;
-      switch (statusCode) {
-        case 400:
-          setToastMessage('The email or password you submited is incorrect.');
-          break;
-        default:
-          setToastMessage('Oops, something went wrong, please try again.');
-      }
-      setLoginIsDisabled(false);
-      setShowToast(true);
-    }
-  }, [login.error]);
+  if (loginMutation.isSuccess) {
+    setSession(loginMutation.data);
+    router.push(Routes.Dashboard);
+    return;
+  }
```

Comment:
Since we don't want to call multiple `router.push`, it is merited to have this in a useEffect (especially now that this is largely simplified)